packages/devtools_app/lib/src/network/network_service.dart

```diff
@@ -39,18 +40,43 @@ class NetworkService {
 
     final timestamp = await serviceManager.service.getVMTimelineMicros();
     final sockets = await _refreshSockets();
-    final timeline = await serviceManager.service.getVMTimeline(
-      timeOriginMicros: networkController.lastRefreshMicros,
-      timeExtentMicros:
-          timestamp.timestamp - networkController.lastRefreshMicros,
-    );
+    List<HttpProfileRequest> httpRequests;
+    Timeline timeline;
+    if (await serviceManager.service.isDartIoVersionSupported(
+      supportedVersion: SemanticVersion(major: 1, minor: 3),
+      isolateId: serviceManager.isolateManager.selectedIsolate.id,
+    )) {
+      httpRequests = await _refreshHttpProfile();
+    } else {
+      timeline = await serviceManager.service.getVMTimeline(
+        timeOriginMicros: networkController.lastRefreshMicros,
+        timeExtentMicros:
+            timestamp.timestamp - networkController.lastRefreshMicros,
+      );
+    }
     networkController.lastRefreshMicros = timestamp.timestamp;
     networkController.processNetworkTraffic(
       timeline: timeline,
       sockets: sockets,
+      httpRequests: httpRequests,
     );
   }
 
+  Future<List<HttpProfileRequest>> _refreshHttpProfile() async {
+    assert(serviceManager.service != null);
+    if (serviceManager.service == null) return [];
+
+    final requests = <HttpProfileRequest>[];
+    await serviceManager.service.forEachIsolate((isolate) async {
+      final request = await serviceManager.service.getHttpProfile(
+        isolate.id,
+        updatedSince: networkController.lastRefreshMicros,
+      );
+      requests.addAll(request.requests);
```

Comment:
can we just return request.requests instead of making a copy of the list?

Reply:
No, since we need to get profiling data from each isolate, which each returns their own `request`.

Reply:
oops - didn't see `forEachIsolate`. sg.